.github/workflows/rustdoc.yml

```diff
@@ -0,0 +1,28 @@
+# This Action triggers a build of Vector's Rustdoc each time a PR is merged
+# to `master`.
+#
+# Netlify account for Rustdoc site: https://app.netlify.com/sites/vector-rustdoc
+# Repo that builds the site: https://github.com/timberio/vector-rustdoc
+# Root URL for Rustdoc site: https://vector-rustdoc.netlify.app
+# Pretty URL for the published docs: https://rustdoc.vector.dev
+name: Publish Vector Rustdoc
+
+# Action is triggered only if changes are merged, not just pushed
+# See also the "if" statement in the job definition
+on:
```

Comment:
👍 this seems like it would work, but I think it'd be more straightforward to just build on pushes to master like the test suite:

https://github.com/timberio/vector/blob/2ac861e09f99036145749ee8af7a7e0d7aa945c6/.github/workflows/test.yml#L5-L7